docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,220 @@
+2023-01-09T07:45:02.563-0800 INFO (substreams-sink-files) reading substreams manifest {"manifest_path": "substreams.yaml"}
+2023-01-09T07:45:02.563-0800 INFO (substreams-sink-files) starting pprof server {"listen_addr": "localhost:6060"}
+2023-01-09T07:45:04.041-0800 INFO (pipeline) computed start block {"module_name": "jsonl_out", "start_block": 0}
+2023-01-09T07:45:04.042-0800 INFO (substreams-sink-files) ready, waiting for signal to quit
+2023-01-09T07:45:04.045-0800 INFO (substreams-sink-files) setting up sink {"block_range": {"start_block": 0, "end_block": "None"}, "cursor": {"Cursor":"","Block":{}}}
+2023-01-09T07:45:04.048-0800 INFO (substreams-sink-files) starting new file boundary {"boundary": "[0, 10000)"}
+2023-01-09T07:45:04.049-0800 INFO (substreams-sink-files) boundary started {"boundary": "[0, 10000)"}
+2023-01-09T07:45:04.049-0800 INFO (substreams-sink-files) starting stats service {"runs_each": "2s"}
+2023-01-09T07:45:06.052-0800 INFO (substreams-sink-files) substreams sink stats {"progress_msg_rate": "0.000 msg/s (0 total)", "block_rate": "650.000 blocks/s (1300 total)", "last_block": "#1299 (a0f0f283e0d297dd4bcf4bbff916b1df139d08336ad970e77f26b45f9a521802)"}
+```
+
+## Conclusion and review
+
+The ability to route data extracted from the blockchain by using Substreams is powerful and useful. Files aren't the only type of sink the data extracted by Substreams can be piped into. Review the core Substreams sinks documentation for [additional information on other types of sinks](https://substreams.streamingfast.io/developers-guide/substreams-sinks) and sinking strategies.
+
+### Recap
+
+To use `substreams-sink-files` you need to clone the official repository, install the tooling, generate the required files from the substreams CLI for the example Substreams module and run the sink tool.
+
+You have to ensure the sinking strategy has been defined, the appropriate file types have been targeted, and accounted for, and the module handler code in your Substreams module has been properly updated. You need to start the `substreams-sink-files` tool and use the `run` command being sure to provide all of the required values for the various flags.
+
+---
+
+**-- QUESTIONS --**
+
+**3. QUESTION**: 'There need to have some content about how the limitation of this sink which write bundles only when last block of a bundle is final.'
+
+<i>I need input on this or some type of reference or something to find information for what we want to tell the reader.</i>
+
+**4. QUESTION**: 'Add discussion about where Substreams cursor is saved and importance of persisting this state (save as a .yaml file).'
+
+<i>I need input on this or some type of reference or something to find information for what we want to tell the reader.</i>
```

Comment:
Substreams uses an opaque `cursor` each time it sends back a block to a consumer. This `cursor` point to the exact location within the blockchain where the block is. If your connection terminates or the process restart, on re-connection, we send back the `cursor` of the last written bundle in the Substreams request so that the stream of data is resumed exactly where we need and data integrity is maintained.

The `cursor` is saved in a file on a disk, the location of this file is specified by the flag `--state-store` which point to a local folder. You **must** ensure that this file is properly save to a persistent location. If the file is lost, the `substreams-sink-files` tool will restart from the beginning of the chain redoing all the previous processing done.

That is why it's important that this file is properly persisted and follows your deployment of `substreams-sink-files`.

Reply:
Added. Thank you.